go/nbs/compacting_chunk_source.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package nbs
+
+import (
+	"sync"
+
+	"github.com/attic-labs/noms/go/d"
+)
+
+func newCompactingChunkSource(mt *memTable, haver chunkReader, p tablePersister, rl chan struct{}) *compactingChunkSource {
+	ccs := &compactingChunkSource{mt: mt}
+	ccs.wg.Add(1)
+	rl <- struct{}{}
+	go func() {
+		defer ccs.wg.Done()
+		var cs chunkSource = emptyChunkSource{}
+		if tableHash, chunkCount := p.Compact(mt, haver); chunkCount > 0 {
```

Comment:
Doesn't need to be this patch, but I think it'd be good to have `Compact` return the `chunkSource`. We always immediately turn around and open it again and it's a bit silly to go read a bunch of bytes from s3 that we just wrote.

Maybe open a bug for that?